### ISSUE TITLE
Exception while profiling '[ActivityListValidator] No sample activity matches the trace'

### ISSUE THREAD

**Issue by MattKotsenas** [microsoft/ApplicationInsights-Profiler-AspNetCore#87]
Hello! I'm getting the following exception when running the profiler

`[ActivityListValidator] No sample activity matches the trace.`

I'm using the following versions:

* Microsoft.ApplicationInsights.AspNetCore 2.14.0 (il:2.14.0-17971)
* Microsoft.ApplicationInsights.Profiler.Core 2.1.0-beta3
* Microsoft.AspnetCore 2.2.0

In the Docker container `mcr.microsoft.com/dotnet/core/aspnet:2.2-stretch-slim`.

```
ServiceProfiler.EventPipe.Client.TraceValidators.ValidateFailedException:
   at ServiceProfiler.EventPipe.Client.TraceValidators.ActivityListValidator.ValidateImp (Microsoft.ApplicationInsights.Profiler.Core, Version=2.6.2004.3001, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at ServiceProfiler.EventPipe.Client.TraceValidators.TraceValidatorBase.Validate (Microsoft.ApplicationInsights.Profiler.Core, Version=2.6.2004.3001, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at ServiceProfiler.EventPipe.Client.TraceValidators.TraceValidatorBase.Validate (Microsoft.ApplicationInsights.Profiler.Core, Version=2.6.2004.3001, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at ServiceProfiler.EventPipe.Client.ServiceProfilerProvider.ValidateSamples (Microsoft.ApplicationInsights.Profiler.Core, Version=2.6.2004.3001, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
```

I wouldn't expect the profiler to be throwing exceptions in any case other than a misconfiguration, as that adds additional overhead to our app and introduces more noise into failure analysis for our app.

Please let me know if there's anything I can provide or do to help reproduce or diagnose this issue.

**Comment by xiaomi7732** [comment]
@MattKotsenas Thanks for the report. Firstly, please help me confirm my understanding:
* This issue doesn't crash your application, however, the fact this error happens generates noise in your logs, correct?
* This happens once a while but this doesn't happen every time profiler run, right?

. The issue you encountered root in the fact that there's a level of suitableness between the Application Insights SDK and the EventPipe events.

If you aren't worry about the functionality of Profiling, a work around FYI is to disable the profiler logs by set profiler logs to level of critical only:
```jsonc
{
  "Logging": {
    "LogLevel": {
      "Default": "Warning", // Keep whatever you have
      "ServiceProfiler": "Critical",  // Only critical issues are logged
      "Microsoft.ServiceProfiler": "Critical"
    }
  }
}
```

On the side, I notice you are using ASP.NET Core 2.2 images. Since .NET Core 2.2 is end of its life cycle, is there reason you don't move the app to .NET Core 3.1?

**Comment by MattKotsenas** [comment]
Hi @xiaomi7732!

Correct, this issue doesn't crash the app, and I will apply the workaround to avoid logging non-critical `ServiceProfiler` events to avoid muddying our logs.

I do still have the concern that throwing exceptions, even if only occasionally, has a negative impact on the performance of our app, which we have to mitigate by reducing our `RandomProfilingOverhead`, which isn't ideal.

Lastly, we're in the process of migrating to dotnet core 3.1 as we speak :).

Let me know if there's any other info I can provide. Thanks!

**Comment by xiaomi7732** [comment]
@MattKotsenas Thank you so much for the effort. We are a bit hand tight here since the .NET Core 2.2 is already at the end of it's support life cycle. That said, The fix for #88 might reduce the chance of this happening.

I'll close this issue for now. Feel free to open new issues when needed.
